### ISSUE TITLE
[New rule] use_internal_library_imports

### ISSUE THREAD

**Issue by lukepighetti** [dart-code-checker/dart-code-metrics#948]
If a there is an internal library and a developer reaches into it, show a linter warning that they should import the library itself.

- [ ] Warns about a potential error (problem)
- [x] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

main.dart
```dart
import 'my_file.dart'; // LINT
import 'my_library.dart'; // NO LINT

void main() {
  print(foo);
}
```

my_file.dart
```dart
const foo = 'foo';
```

my_library.dart
```dart
library my_library;

export 'my_file.dart';
```

**Are you willing to submit a pull request to implement this rule?**

Not at this time
